### PR TITLE
Add basic ann tutorial.

### PR DESCRIPTION
**Pull request by zoq**
This is a basic Neural Network tutorial, that describes the existing framework from a higher level. For details, we should point to the layer-specific documentation.

### CODE REVIEW

File: doc/tutorials/ann/ann.txt [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+  model.Add<Linear<> >(4, 10);
+  model.Add<LogSoftMax<> >();
+
+  StandardSGD opt(0.1, 1, input.n_cols /* 1 epoch */, -100);
+  model.Train(input, labels, opt);
+}
+@endcode
+
+For further examples on the ann usage of the ann classes, see [mlpack
+models](https://github.com/mlpack/models)
```
Comment (rcurtin) [inline doc/tutorials/ann/ann.txt]: This is a tiny issue but probably we should put a period at the end of this sentence.  I think "ann usage" should be just "usage" also. :)

File: doc/tutorials/ann/ann.txt [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+}
+@endcode
+
+For further examples on the usage of the ann classes, see [mlpack
+models](https://github.com/mlpack/models).
+
+@section layer_api_anntut Layer API
+
+In order to facilitate consistent implementations, we have defined a LayerType
+API that describes all the methods that a Layer may implement. mlpack offers a
```
Comment (ShikharJ) [inline doc/tutorials/ann/ann.txt]: `layer` in place of `Layer` here and at other places (4 in total)?

File: doc/tutorials/ann/ann.txt [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
+implementation of a \c Forward() method. The interface looks like:
+
+@code
+template<typename eT>
+void Forward(const arma::Mat<eT>&& input, arma::Mat<eT>&& output);
+@endcode
+
+The method should calculate the output of the layer given the input matrix and
+store the result in the given output matrix. Next, any Layer must implement the
+Backward() method, which uses certain computations obtained during forward and
```
Comment (ShikharJ) [inline doc/tutorials/ann/ann.txt]: `Forward()` in place of `forward`?

File: doc/tutorials/ann/ann.txt [hunk 4]
```diff
@@ -0,0 +1,673 @@
...
+should calculate the function f(x) by propagating x backward trough f:
```
Comment (ShikharJ) [inline doc/tutorials/ann/ann.txt]: `through` in place of `trough`?

File: doc/tutorials/ann/ann.txt [hunk 5]
```diff
@@ -0,0 +1,673 @@
...
+  {
+    /* Nothing to do here */
+  }
+}
+@endcode
+
+The constructor for \c ExampleLayer will build the layer given the input and
+output size. Note that, if the input or output size information isn't used
+internally it's not necessary to provide a specific constructor. Also, one could
+add additional or other information that are necessary for the layer
```
Comment (ShikharJ) [inline doc/tutorials/ann/ann.txt]: `that is` in place of `that are`?

File: doc/tutorials/ann/ann.txt [hunk 6]
```diff
@@ -0,0 +1,673 @@
...
+              GradientType&& gradient)
+{
+  gradient = arma::zeros(input.n_rows, input.n_cols) * error;
+}
+@endcode
+
+The three functions \c Forward(), \c Backward() and \c Gradient() (which is
+needed for a differentiable layer) contain the main logic of the layer. The
+following functions are just to access and manipulate the different layer
+parameter.
```
Comment (ShikharJ) [inline doc/tutorials/ann/ann.txt]: `parameters`?

File: doc/tutorials/ann/ann.txt (hunk 5 above)
Comment (zoq) [reply doc/tutorials/ann/ann.txt]: I think both works here.

**Comment by zoq** [comment]
Thanks for the reviews, I will go ahead and merge.

**Pull request merged**